packages/shared/src/components/post/write/CreatePostButton.tsx

```diff
@@ -43,20 +44,34 @@ export function CreatePostButton({
     return !allowedToPost;
   };
 
-  return (
+  const href =
+    link.post.create + (squad && allowedToPost ? `?sid=${squad.handle}` : '');
+
+  const renderBtn = () => (
     <Button
       variant={ButtonVariant.Secondary}
       className={className}
       disabled={getIsDisabled()}
-      icon={compact && !isLaptop && <PlusIcon />}
+      icon={compact && <PlusIcon />}
       tag="a"
-      href={
-        link.post.create +
-        (squad && allowedToPost ? `?sid=${squad.handle}` : '')
-      }
       size={isLaptop ? ButtonSize.Medium : ButtonSize.Small}
+      {...(!compact && {
+        tag: 'a',
+        href,
+      })}
     >
-      {!compact || isLaptop ? 'New post' : null}
+      {!compact ? 'New post' : null}
     </Button>
   );
+
+  return compact ? (
+    <LinkWithTooltip
```

Comment:
Wouldn't it be better with conditional wrapper component? This way we can avoid the renderBtn() at all

Reply:
Actually we can just render and have the tooltip disabled when it's compact :) (as another alternative)
@omBratteng Just added a feature for this.

Reply:
See [here](https://github.com/dailydotdev/apps/pull/2714/files#diff-6eafb5899cb23318abfef2ff71f5bf4f40e4bfd3c2b00de80adf904539ae59e9) for how I did it. (add `show` prop)

```diff
@@ -13,13 +13,14 @@ const TippyTooltip = dynamicParent<TooltipProps>(
 );
 
 export function SimpleTooltip({
+  show = true,
   children,
   content,
   onTrigger,
   onShow,
   forceLoad,
   ...props
-}: TooltipProps): ReactElement {
+}: TooltipProps & { show?: boolean }): ReactElement {
   /**
    * We introduced the `shouldShow` variable to manage a re-focus issue
    * The old implementation would re-focus the tooltip whenever a modal would close
@@ -64,6 +65,10 @@ export function SimpleTooltip({
     return false;
   };
 
+  if (!show) {
+    return component;
+  }
+
   return (
     <TippyTooltip
       {...props}
@@ -78,13 +83,4 @@ export function SimpleTooltip({
   );
 }
```

Reply:
Since we are not using `SimpleTooltip` here, I went ahead with the wrapper approach, I kind of prefer it better too

Reply:
Wrapper is fine in this case 👍
Just note it can cause some re render issues if you toggle evaluations during render. (Shouldnt apply here i think)

Reply:
Could add `show` prop to `LinkWithTooltip`, and if that is false, you just return the `component`

Reply:
I dont see any commits that change this is that correct?

Reply:
I don't see any commits either? 🤔

Reply:
wnot sure what happened, this should be updated now